src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -218,14 +218,27 @@ private synchronized void updateSubscriptionsInternal() {
         Set<String> topicsToSubscribe = new HashSet<>(toSubscribeLocalMqttTopics);
         topicsToSubscribe.removeAll(subscribedLocalMqttTopics);
 
-        // TODO: Support configurable qos, add retry
+        // TODO: Support configurable qos
         topicsToSubscribe.forEach(s -> {
-            try {
-                mqttClientInternal.subscribe(s);
-                LOGGER.atDebug().kv(TOPIC, s).log("Subscribed to topic");
-                subscribedLocalMqttTopics.add(s);
-            } catch (MqttException e) {
-                LOGGER.atError().kv(TOPIC, s).log("Failed to subscribe");
+            int waitBeforeRetry = MIN_WAIT_RETRY_IN_SECONDS;
+            while (!Thread.currentThread().isInterrupted()) {
+                try {
+                    mqttClientInternal.subscribe(s);
```

Comment:
You've only added retries for the mqtt client which connects to the local broker. You must add this for the Greengrass IPC client, that's actually what I wanted originally.

Reply:
+1

Reply:
Could you add a bit more context please? Do you mean retry subscribe requests for PubSub client:
https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/4457827a7c5b4655247dc5c60d3ff49ea5595074/src/main/java/com/aws/greengrass/mqttbridge/clients/PubSubClient.java#L125-L127

`pubSubIPCAgent.subscribe()` here does not throw any checked exceptions. Do we want to add retries on generic `Exception`?

Reply:
https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/ca8ea69e0bb8734b1b40072a627ff8de63a76919/src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java#L133

Reply:
Thanks, added retry for IoTCoreClient as well.